src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -2749,16 +2756,8 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 		return true;
 	}
 
-	private async _ensureWorkspaceTasks(): Promise<void> {
-		if (!this._workspaceTasksPromise) {
-			await this.getWorkspaceTasks();
-		} else {
-			await this._workspaceTasksPromise;
-		}
-	}
-
 	private async _runTaskCommand(filter?: string | ITaskIdentifier): Promise<void> {
-		await this._ensureWorkspaceTasks();
+		await this._waitForTaskSystem();
 		if (!filter) {
```

Comment:
Are you sure we don't need to await both the task system and `_ensureWorkspaceTasks`?

Reply:
well, `getWorkspaceTasks` now waits for the task system. It's fine without that call based on my testing, but probably safer to `ensureWorkspaceTasks` anyway.